src/main/java/org/apache/mesos/executor/HealthCheckHandler.java

```diff
@@ -0,0 +1,177 @@
+        double delay = taskInfo.getHealthCheck().getDelaySeconds() +
+                taskInfo.getHealthCheck().getGracePeriodSeconds();
+
+        long intervalMs = (long) (interval * 1000);
+        long delayMs = (long) (delay * 1000);
+
+        logger.info("Scheduling health check.");
+        return scheduledExecutorService.scheduleAtFixedRate(
+                new HealthCheckRunner(taskInfo.getHealthCheck(), healthCheckStats),
+                delayMs,
+                intervalMs,
+                TimeUnit.MILLISECONDS);
+    }
+
+    private void validateTask(Protos.TaskInfo taskInfo) throws HealthCheckValidationException {
+        // Validate TaskInfo
+        if (!taskInfo.hasHealthCheck()) {
+            throw new HealthCheckValidationException("The following task does not contain a HealthCheck: " + taskInfo);
+        }
+
+        // Validate HealthCheck
+        Protos.HealthCheck healthCheck = taskInfo.getHealthCheck();
+        if (healthCheck.hasHttp()) {
+            throw new HealthCheckValidationException(
+                    "The following task contains an unsupported HTTP HealthCheck: " + taskInfo);
+        }
+
+        if (!healthCheck.hasCommand()) {
+            throw new HealthCheckValidationException(
+                    "The following task does not contain a Command for its Healthcheck: " + taskInfo);
+        }
+
+        // Validate Command
+        Protos.CommandInfo commandInfo = healthCheck.getCommand();
+        if (!Boolean.valueOf(commandInfo.getShell())) {
+            throw new HealthCheckValidationException(
+                    "Only shell based health checks are supported for health check commmand: " + commandInfo);
+        }
+    }
+
+    /**
+     * This class spawns a subprocess for each invocation of a health check.  It records statistics
+     * regarding successes and failures.
+     */
+    private static class HealthCheckRunner implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(getClass());
+        private final Protos.HealthCheck healthCheck;
+        private final HealthCheckStats healthCheckStats;
+
+        public HealthCheckRunner(Protos.HealthCheck healthCheck, HealthCheckStats healthCheckStats) {
+            this.healthCheck = healthCheck;
+            this.healthCheckStats = healthCheckStats;
+            logger.info("Created health check runner.");
+        }
+
+        @Override
+        public void run() {
+            logger.info("Running health check");
+            Protos.CommandInfo commandInfo = healthCheck.getCommand();
+
+            try {
+                final Map<String, String> envMap = TaskUtils.fromEnvironmentToMap(commandInfo.getEnvironment());
+                List<String> command = new ArrayList<>();
+                command.add("/bin/bash");
+                command.add("-c");
+                command.add(commandInfo.getValue());
+
+                String[] cmdArray = new String[command.size()];
```

Comment:
Doesn't compile. `Error:(134, 53) java: variable cmdArray might not have been initialized`